### PR TITLE
Add kwargs for flow_executor creation and use it for name

### PR DESCRIPTION
**Pull request by huaiyan**
# Description
For otel trace, we need to set name field.
Before:
For flex, we use function name directly.
For flow, we use name section from yaml or working dir's folder name.

That's ok for local execution, since customer could see their folder name easily.

But for runtime execution, using folder name may not work, since there is display name concept.
So, add kwargs in flow_executor `create` method, and change flow's name accordingly.
**Name priority: name from yaml payload > name from content > working_dir.stem**
For portal, there is a meaningless predefined name in yaml, we should use name from payload to override it.

Will open another PR for multi container's scenario.

# All Promptflow Contribution checklist:
- [X] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-core/promptflow/contracts/flow.py [hunk 1]
```diff
@@ -702,12 +702,13 @@ def _update_working_dir(cls, working_dir: Path):
         sys.path.insert(0, str(working_dir))
 
     @classmethod
-    def from_yaml(cls, flow_file: Path, working_dir=None) -> "Flow":
+    def from_yaml(cls, flow_file: Path, working_dir=None, fallback_name=None) -> "Flow":
```
Comment (thy09) [inline src/promptflow-core/promptflow/contracts/flow.py:705]: fallback_name is not a good name.
Maybe just `name` and declare the priorities here

Comment (huaiyan) [reply src/promptflow-core/promptflow/contracts/flow.py]: Fixed.

**Review by thy09 — APPROVED** [review]

File: src/promptflow-core/promptflow/contracts/flow.py [hunk 2]
```diff
@@ -702,12 +702,14 @@ def _update_working_dir(cls, working_dir: Path):
...
 
     @classmethod
-    def from_yaml(cls, flow_file: Path, working_dir=None) -> "Flow":
+    def from_yaml(cls, flow_file: Path, working_dir=None, name=None) -> "Flow":
         """Load flow from yaml file."""
         working_dir = cls._parse_working_dir(flow_file, working_dir)
         with open(working_dir / flow_file, "r", encoding=DEFAULT_ENCODING) as fin:
             flow_dag = load_yaml(fin)
-        flow_dag["name"] = flow_dag.get("name", _sanitize_python_variable_name(working_dir.stem))
+        # Name priority: name from yaml content > name from payload > working_dir.stem
```
Comment (riddlexu) [inline src/promptflow-core/promptflow/contracts/flow.py:710]: Maybe the name from payload should have the highest priority?

Comment (huaiyan) [reply src/promptflow-core/promptflow/contracts/flow.py]: Good catch, change the priority.

**Review by thy09 — APPROVED** [review]

**Review by riddlexu — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Pull request merged**